src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -412,6 +434,36 @@ def bucket_exists(self, bucket_name, userid, request=None):
             raise e
 
     @RestClient.api_put('/{bucket_name}')
-    def create_bucket(self, bucket_name, request=None):
-        logger.info("Creating bucket: %s", bucket_name)
-        return request()
+    def create_bucket(self, bucket_name, zonegroup, placement_target, request=None):
+        logger.info("Creating bucket: %s, zonegroup: %s, placement_target: %s",
+                    bucket_name, zonegroup, placement_target)
+        return request(data=''.join((
+            '<CreateBucketConfiguration>',
+            '<LocationConstraint>{}:{}</LocationConstraint>'.format(zonegroup, placement_target),
+            '</CreateBucketConfiguration>'
```

Comment:
Would it make sense to use a library (`dictoxml`?) to convert Python objects?

Reply:
I personally don't find it necessary as we are sending the constraint as body string,
but I'm open to discuss this.

Reply:
Ok. When there's some (de)serialization I always prefer to use library, as they usually handle potential escaping, validation and injection issues.

Reply:
> Ok. When there's some (de)serialization I always prefer to use library, as they usually handle potential escaping, validation and injection issues.

I implemented a more formal approach using ```xml.etree.ElementTree``` API to generate XML string without introducing additional dependency.